pymbar/mbar_solvers.py

```diff
@@ -301,51 +447,58 @@ def adaptive(u_kn, N_k, f_k, tol=1.0e-12, options=None):
     # to calculate the first time.
     g = mbar_gradient(u_kn, N_k, f_k)  # Objective function gradient.
 
-    for iteration in range(0, options["maximum_iterations"]):
+    maxiter = options["maxiter"]
+    min_sc_iter = options["min_sc_iter"]
+    warn = "Did not converge."
+    for iteration in range(0, maxiter):
```

Comment:
you can actually `jit` a loop using `jax.lax.while_loop`:

https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.while_loop.html`

Reply:
(if the number of iterations is known before hand, `jax.lax.fori_loop` is another option, but probably doesn't apply hear since we break adaptively)